src/protocol/primitives/record-batch-attributes.ts

```diff
@@ -0,0 +1,107 @@
+import { IllegalArgumentError } from '../exceptions';
+import { type Int16 } from './int16';
+
+export class Compression {
+  public static readonly None = 0b00000000;
+  public static readonly GZIP = 0b00000001;
+  public static readonly Snappy = 0b00000010;
+  public static readonly LZ4 = 0b00000011;
+  public static readonly ZSTD = 0b00000100;
+  private static readonly MASK = 0b00000000_00000111;
+
+  public static fromInt16({ value }: Int16): CompressionValue {
+    const compression = value & this.MASK;
+    if (compression > 4) {
+      throw new IllegalArgumentError(`Expected compression value to be in range 0-4, received ${value}`);
+    }
+
+    return compression as CompressionValue;
+  }
+}
+
+export type CompressionValue =
+  | typeof Compression.None
+  | typeof Compression.GZIP
+  | typeof Compression.Snappy
+  | typeof Compression.LZ4
+  | typeof Compression.ZSTD;
+
+export class TimestampType {
+  public static readonly CreateTime = 0b00000000;
+  public static readonly LogAppendTime = 0b00001000;
+  private static readonly MASK = 0b00000000_00001000;
+
+  public static fromInt16({ value }: Int16): TimestampTypeValue {
+    const timestampType = value & this.MASK;
+
+    if (timestampType & TimestampType.LogAppendTime) {
+      return TimestampType.LogAppendTime;
+    }
+
+    return TimestampType.CreateTime;
+  }
+}
+
+export type TimestampTypeValue = typeof TimestampType.CreateTime | typeof TimestampType.LogAppendTime;
+
+export class Transactional {
+  public static readonly No = 0b00000000;
+  public static readonly Yes = 0b00010000;
+  private static readonly MASK = 0b00000000_00010000;
+
+  public static fromInt16({ value }: Int16): TransactionalValue {
+    const transactional = value & this.MASK;
+
+    if (transactional & Transactional.Yes) {
```

Comment:
Can it be simplified to the `transactional !== 0`?

Reply:
Maybe I will just compare it with `Transactional.Yes`?

Reply:
Sounds even better.

Reply:
Now when I'm thinking about it, after performing AND with MASK, we could just return the value as in Compression. What do you think?

Reply:
The simpler the better 😅